packages/web-runtime/tests/components/Notification.spec.js

```diff
@@ -0,0 +1,235 @@
+
+config.showDeprecationWarnings = false
+
+const selectors = {
+  notificationBell: '#oc-notification-bell',
+  ocIconStub: 'oc-icon-stub',
+  actionButton: '.oc-ml-s',
+  resolveNotificationButton: '#resolve-notification-button',
+  link: '.uk-link',
+  subject: 'h4',
+  message: '.uk-text-small'
+}
+
+const defaultNotificationData = {
+  notification_id: 1,
+  subject: 'Test',
+  message: 'This is a test message',
+  actions: []
+}
+
+function getStoreWithActiveNotifications(activeNotifications) {
+  return new Store({
+    getters: {
+      activeNotifications: activeNotifications
+    }
+  })
+}
+
+function getStoreWithEmptyActions() {
+  return getStoreWithActiveNotifications([
+    {
+      ...defaultNotificationData
+    }
+  ])
+}
+
+function getStoreWithSingleAction() {
+  return getStoreWithActiveNotifications([
+    {
+      ...defaultNotificationData,
+      actions: [
+        {
+          label: 'Test Action'
+        }
+      ]
+    }
+  ])
+}
+
+function getStoreWithTwoActiveNotifications() {
+  return getStoreWithActiveNotifications([
+    {
+      subject: 'First Notification',
+      message: 'This is first test message',
+      actions: []
+    },
+    {
+      subject: 'Second Notification',
+      message: 'This is second test message',
+      actions: []
+    }
+  ])
+}
+
+function getStoreWithLinkActions() {
+  return getStoreWithActiveNotifications([
+    {
+      link: 'http://test.link',
+      actions: []
+    }
+  ])
+}
+
+describe('Notification component', () => {
+  describe('notification bell', () => {
+    const store = getStoreWithEmptyActions()
+    const mocks = {
+      $store: store
+    }
+    let wrapper
+    beforeEach(() => {
+      store.reset()
+      wrapper = shallowMount(Notifications, { stubs, mocks })
+    })
+    it('displays notification bell', () => {
+      const bell = wrapper.find(selectors.notificationBell)
+      expect(bell.exists()).toBeTruthy()
+      expect(bell.find(selectors.ocIconStub).exists()).toBeTruthy()
+      expect(bell.find(selectors.ocIconStub).attributes('name')).toBe('bell')
+      expect(bell.attributes('uk-tooltip')).toBe('Notifications')
+      expect(bell.attributes('aria-label')).toBe('Notifications')
+    })
+  })
+  describe('when active notification contains a message', () => {
+    const store = getStoreWithEmptyActions()
+    const mocks = {
+      $store: store
+    }
+    let wrapper
+    beforeEach(() => {
+      store.reset()
+      wrapper = shallowMount(Notifications, { stubs, mocks })
+    })
+    it('displays the notification message', () => {
+      const messageElement = wrapper.find(selectors.message)
+      expect(messageElement.exists()).toBeTruthy()
+      expect(messageElement.text()).toBe('This is a test message')
+    })
+    it('displays the notification subject', () => {
+      const subjectElement = wrapper.find(selectors.subject)
+      expect(subjectElement.exists()).toBeTruthy()
+      expect(subjectElement.text()).toBeTruthy()
```

Comment:
can we also check for the subject content?

Reply:
yes